backend/src/main/controller/EnumController.ts

```diff
@@ -0,0 +1,64 @@
+import { DiscountType } from '../data/DiscountType';
+import { PromotionCategory } from '../data/PromotionCategory';
+import { CuisineType } from '../data/CuisineType';
+import { NextFunction, Request, Response } from 'express';
+import Joi from 'joi';
+
+enum SupportedEnums {
+  DiscountType = 'DiscountType',
+  PromotionCategory = 'PromotionCategory',
+  CuisineType = 'CuisineType',
+}
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+export class EnumController {
+  // validates that incoming string is of type SupportedEnums
+  private enumSchemaValidation = Joi.string()
+    .valid(...Object.values(SupportedEnums))
+    .required();
+
+  /**
+   * returns all possible values of an enum. Otherwise throws validation error if not a supported enum
+   * */
+  getEnum = async (
+    request: Request,
+    response: Response,
+    next: NextFunction
+  ): Promise<any> => {
+    try {
+      const enumString: SupportedEnums = await this.enumSchemaValidation.validateAsync(
+        request.params.enum,
+        { abortEarly: false }
+      );
+      const enumClass = this.getEnumClass(enumString);
+      const enumValues = this.getEnumValues(enumClass);
+      return response.status(200).send(enumValues);
+    } catch (e) {
+      next(e);
+    }
+  };
+
+  /**
+   * returns the corresponding class matching the enumString
+   * */
+  private getEnumClass(supportedEnums: SupportedEnums) {
+    switch (supportedEnums) {
```

Comment:
Unfortunately had to use switch statement. Does anyone have any suggestions? Typescript won't let me map to type `enum`.

I basically want to map a string to an `enum`

Reply:
I can't think of anything that would be a much better alternative :( You could drop the enclosing brackets for each case to make it a little less verbose though

Reply:
Ohya for sure